src/app/_components/main-card.tsx

```
@@ -16,13 +16,17 @@ function Card({
+
+  const handleClick = () => {
+    const queryParams = new URLSearchParams(query).toString();
+    router.push(`/detail/${id}${queryParams ? `?${queryParams}` : ""}`);
+  };
+
   return (
-    <div
-      className="relative flex h-30 w-full gap-3"
-      onClick={() => router.push(`/detail/${id}`)}
-    >
+    <div className="relative flex h-30 w-full gap-3" onClick={handleClick}>
```

Biome (2.1.2)

[error] Static Elements should not be interactive.

To add interactivity such as a mouse or key event listener to a static element, give the element an appropriate role value.

(lint/a11y/noStaticElementInteractions)